### DISCUSSION TITLE
Obstacle avoidance

### DISCUSSION THREAD

**Discussion by sushmithabhagya** [ctu-mrs/mrs_uav_system#144] [Q&A]
Hello,
I have recently installed Mrs_UAV_system and working on UAV formation control and Obstacle avoidance. Could you please tell me how to incorporate mrs bumper to avoidance obstacle while following a trajectory path.

**Comment by klaxalk** [comment]
Hey, I would not recommend to use the mrs_bumper for that. The bumper is more of a last resort safety feature, which might be more harmful than useful for trajectory following. Instead, check the https://github.com/ctu-mrs/octomap_mapping_planning metapackage. That contains on possible solution to the problem. We are building a 3D occupancy map using the OctoMap library. That is used for path planning. While flying along the planner path, a predicted future trajectory (coming out of the MpcTracker) is checked against the map to spot new obstacles and the tracking is stopped if collision is predicted. This is import to do independently on the path planning, since the planner can take up to several seconds to replan, during which the UAV can easily move many meters.
